### PR TITLE
drivers: sensor: bq274xx: fix lazy loading

### LINKED ISSUE #47458: BQ274XX Sensors Driver - Fails with CONFIG_BQ274XX_LAZY_CONFIGURE
**Describe the bug**
When using the BQ274XX Sensors driver, there is a kconfig CONFIG_BQ274XX_LAZY_CONFIGURE that allows the device not to be imitated until it's first use. I use this as the communication pull ups in the BQ274XX are disabled on boot (enabled in boot process however).

If CONFIG_BQ274XX_LAZY_CONFIGURE  is set however, compilation fails as the following code references a bq274xx struct that doesn't exist. See lines 405-409 of ${zephyr}/drivers/sensor/bq274xx/bq274xx.c
```
#ifdef CONFIG_BQ274XX_LAZY_CONFIGURE
	bq274xx->lazy_loaded = false;
#else
	status = bq274xx_gauge_configure(dev);
#endif
```

**To Reproduce**
I'd have thought you should be able to compile the sample ${zephyr}/samples/sensor/bq274xx/ but with the CONFIG_BQ274XX_LAZY_CONFIGURE  set to get the error

**Expected behavior**
 CONFIG_BQ274XX_LAZY_CONFIGURE should allow the BQ274XX not to be imitated until it's first use

**Impact**
Compilation failure

**Logs and console output**
```
F:/zephyrproject/zephyr/drivers/sensor/bq274xx/bq274xx.c: In function 'bq274xx_gauge_init':
F:/zephyrproject/zephyr/drivers/sensor/bq274xx/bq274xx.c:406:2: error: 'bq274xx' undeclared (first use in this function)
  406 |  bq274xx->lazy_loaded = false;
```

**Environment (please complete the following information):**
 - Windows 10
 - Zephyr with GCC
 - zephyr-v3.1.0-1585-ga83cec9f8b

### PR DESCRIPTION
**Pull request by pdgendt**
This change allows per-instance configuration of lazy loading and fixes build issue.

Fixes #47458

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
rename dts property.

**Review by MaureenHelm — DISMISSED** [review]

File: dts/bindings/sensor/ti,bq274xx.yaml [hunk 1]
```diff
@@ -40,3 +40,9 @@ properties:
         specific events happening (e.g. change in State of Charge). While in
         shutdown mode it acts as an input and toggling it will make the sensor
         exit the mode.
+
+    zephyr,lazy-load:
+      type: boolean
+      description: |
+        Configuring the sensor can take a long time, using lazy loading we can delay
+        till the first sample request and keep the boot time as short as possible.
```
Comment (thedjnK) [inline dts/bindings/sensor/ti,bq274xx.yaml:48]: *until

**Review by thedjnK — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by pdgendt** [comment]
@thedjnK do you approve with the change? (I can't seem to click the re-request review button)

**Review by thedjnK — APPROVED** [review]

**Comment by pdgendt** [comment]
> @galak please revisit

:pray:

**Review by galak — APPROVED** [review]

**Pull request merged**